src/network/rpc/eth/impl/Watches.hpp

```diff
@@ -0,0 +1,193 @@
+#pragma once
+
+#include <chrono>
+#include <queue>
+
+#include "LogFilter.hpp"
+#include "data.hpp"
+
+namespace taraxa::net::rpc::eth {
+using namespace chrono;
+
+// TODO taraxa simple exception
+DEV_SIMPLE_EXCEPTION(WatchLimitExceeded);
+
+using WatchID = uint64_t;
+
+inline auto const watch_id_type_mask_bits = WatchType(ceil(log2(float(COUNT))));
+
+struct placeholder_t {};
+template <WatchType type_, typename InputType_, typename OutputType_ = placeholder_t, typename Params = placeholder_t>
+struct WatchGroup {
```

Comment:
why do you use struct instead of class for such relatively complex data stucture ?

Reply:
The only difference between class and struct in C++ is that class has private visibility by default and struct - public.

I use class or struct keyword only with this in mind. For instance, if I want the first definitions in a class/struct be private - I will use `class`. Otherwise - `struct`

Reply:
This is more about conventions. I will quote Stroustrup: "structs are primarily used for data structures where the members can take any value; that is, we can't define any meaningful invariant "  - you can find this in book "Principles and Practice".

In other words struct in modern C++ are used mostly for short data structures where all members are public, someone uses it purely for POD's, etc....

Software engineering is also about making life easier to other devs so if there is sort of 'rule' in cpp community and people are used to it, lets stick to that.

Reply:
done